packages/mdc-snackbar/mdc-snackbar.scss

```diff
@@ -21,152 +21,183 @@
 //
 
 @import "@material/animation/functions";
+@import "@material/button/mixins";
+@import "@material/icon-button/mixins";
+@import "@material/ripple/mixins";
 @import "@material/rtl/mixins";
+@import "@material/theme/functions";
 @import "@material/theme/mixins";
+@import "@material/theme/variables";
 @import "@material/typography/mixins";
+@import "./mixins";
 @import "./variables";
 
 // postcss-bem-linter: define snackbar
 .mdc-snackbar {
+  @include mdc-snackbar-z-index($mdc-snackbar-z-index);
+  @include mdc-snackbar-viewport-margin($mdc-snackbar-viewport-margin-narrow);
+
   display: flex;
   position: fixed;
+  right: 0;
   bottom: 0;
-  left: 50%;
+  left: 0;
   align-items: center;
-  justify-content: flex-start;
+  justify-content: center;
   box-sizing: border-box;
-  padding-right: 24px;
-  padding-left: 24px;
-  transform: translate(-50%, 100%);
-  transition: mdc-animation-exit-permanent(transform, .25s);
-  background-color: $mdc-snackbar-background-color;
-  pointer-events: none;
-  will-change: transform;
 
-  @media (max-width: ($mdc-snackbar-tablet-breakpoint - 1)) {
-    left: 0;
-    width: 100%;
-    transform: translate(0, 100%);
-  }
+  // Prevent snackbar from A) receiving mouse events while closed, and B) being visible during elastic scrolling.
+  // We can't use `display: none` on the root or surface elements because it prevents CSS transitions from animating.
+  transform: translateY(200vh);
 
-  @media (min-width: $mdc-snackbar-tablet-breakpoint) {
-    min-width: 288px;
-    max-width: 568px;
-    border-radius: 2px;
-  }
-}
-
-.mdc-snackbar--align-start {
-  @media (min-width: $mdc-snackbar-tablet-breakpoint) {
-    @include mdc-rtl-reflexive-position(left, 24px);
-
-    bottom: 24px;
-    transform: translate(0, 200%);
-  }
+  // Ignore mouse events on the root layout element.
+  pointer-events: none;
 
-  @media (max-width: ($mdc-snackbar-tablet-breakpoint - 1)) {
-    bottom: 0;
-    left: 0;
-    width: 100%;
-    transform: translate(0, 100%);
-  }
+  // Mobile Safari only supports `rgba` values for this property; named values like `transparent` are ignored.
+  // From Apple's docs:
+  // > This property obeys the alpha value, if specified.
+  // > If you don’t specify an alpha value, Safari on iOS applies a default alpha value to the color.
+  // > To disable tap highlighting, set the alpha value to 0 (invisible).
+  // > If you set the alpha value to 1.0 (opaque), the element is not visible when tapped.
+  // See https://github.com/ben-eb/postcss-colormin/issues/1
+  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
```

Comment:
Why is this needed on the entire snackbar element?

Reply:
For some reason, iOS Safari displays a tap highlight on the entire snackbar element 🤷‍♀️